### PR TITLE
Add xgboost

### PR DESCRIPTION
**Pull request by nerdcha**
- The xgboost package is popular
 - It's under rapid development
 - This seems to be the simplest way to install it from github

### CODE REVIEW

File: python/Dockerfile [hunk 1]
```diff
@@ -7,3 +7,8 @@ RUN conda install pip statsmodels seaborn python-dateutil nltk -y && \
     # set backend for matplotlibrc to Agg
 RUN matplotlibrc_path=$(python -c "import site, os, fileinput; packages_dir = site.getsitepackages()[0]; print(os.path.join(packages_dir, 'matplotlib', 'mpl-data', 'matplotlibrc'))") && \
     sed -i 's/^backend      : Qt4Agg/backend      : Agg/' $matplotlibrc_path
+
+    # Install bleeding-edge xgboost
+RUN apt-get update && apt-get install -y git && apt-get install -y build-essential \
+    && cd && git clone https://github.com/dmlc/xgboost.git && cd xgboost && make \
```
Comment (moserware) [inline python/Dockerfile]: Do you intentionally want to clone to ~? Probably OK

File: python/Dockerfile [hunk 2]
```diff
@@ -7,3 +7,8 @@ RUN conda install pip statsmodels seaborn python-dateutil nltk -y && \
     # set backend for matplotlibrc to Agg
 RUN matplotlibrc_path=$(python -c "import site, os, fileinput; packages_dir = site.getsitepackages()[0]; print(os.path.join(packages_dir, 'matplotlib', 'mpl-data', 'matplotlibrc'))") && \
     sed -i 's/^backend      : Qt4Agg/backend      : Agg/' $matplotlibrc_path
+
+    # Install bleeding-edge xgboost
+RUN apt-get update && apt-get install -y git && apt-get install -y build-essential \
```
Comment (moserware) [inline python/Dockerfile]: This is OK for the short term, but be advised that each time you do a RUN, you create a separate Docker filesystem layer that has some overhead to it. You might consider grouping all the apt-gets in this file together.

**Comment by moserware** [comment]
:corn:

File: python/Dockerfile (hunk 1 above)
Comment (nerdcha) [inline python/Dockerfile]: Would /usr/local/xgboost be more natural?

File: python/Dockerfile (hunk 2 above)
Comment (nerdcha) [inline python/Dockerfile]: Right, ok, will do.

File: python/Dockerfile (hunk 1 above)
Comment (moserware) [inline python/Dockerfile]: Maybe /usr/local/lib/xgboost ?

Comment (moserware) [inline python/Dockerfile]: or maybe /usr/local/src/xgboost is better

**Pull request merged**